### ISSUE TITLE
Missed incoming User-Plane PRACH messages

### ISSUE THREAD

**Issue by ehitimum** [srsran/srsRAN_Project#504]
## Issue Description ##
After the `2024-03-12T05:36:04.692031 [GNB     ] [I] Radio Unit started successfully`, I started getting these error untill I stop my gnb.
```
2024-03-12T05:36:04.706010 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '477.17' and sector#0
2024-03-12T05:36:04.707012 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '477.19' and sector#0
2024-03-12T05:36:04.711014 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '478.7' and sector#0
2024-03-12T05:36:04.712014 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '478.9' and sector#0
2024-03-12T05:36:04.716017 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '478.17' and sector#0
2024-03-12T05:36:04.717008 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '478.19' and sector#0
2024-03-12T05:36:04.721015 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '479.7' and sector#0
2024-03-12T05:36:04.722011 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '479.9' and sector#0
2024-03-12T05:36:04.726011 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '479.17' and sector#0
2024-03-12T05:36:04.727013 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '479.19' and sector#0
2024-03-12T05:36:04.731020 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '480.7' and sector#0
2024-03-12T05:36:04.732009 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '480.9' and sector#0
2024-03-12T05:36:04.736008 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '480.17' and sector#0
```

## Setup Details ##
This is my complete log file, 
[test-gnb.log](https://github.com/srsran/srsRAN_Project/files/15051171/test-gnb.log)
and this another log where I captured the log with much detail:
[gnb1.zip](https://github.com/srsran/srsRAN_Project/files/14568424/gnb1.zip)
(please uzip before reading it.)
I have a grand master that is connected with cu/du machine and ru machine with a front haul switch. I have tried many different prach configuration. Currently prach configuration index is 165.

This is the gnb configuration file,  
```
amf:
  addr: 192.168.88.10                                               # The address or hostname of the AMF.
  bind_addr: 192.168.88.233                                            # A local IP that the gNB binds to for traffic from the AMF.

ru_ofh:
  ru_bandwidth_MHz: 100                                           # RU instantaneous bandwidth.
  t1a_max_cp_dl: 500                                              # Maximum T1a on Control-Plane for Downlink in microseconds.
  t1a_min_cp_dl: 250                                              # Minimum T1a on Control-Plane for Downlink in microseconds.
  t1a_max_cp_ul: 465                                              # Maximum T1a on Control-Plane for Uplink in microseconds.
  t1a_min_cp_ul: 250                                              # Minimum T1a on Control-Plane for Uplink in microseconds.
  t1a_max_up: 250                                                 # Maximum T1a on User-Plane in microseconds.
  t1a_min_up: 80                                                  # Minimum T1a on User-Plane in microseconds.
  ta4_max: 500                                                    # Maximum Ta4 on User-Plane in microseconds.
  ta4_min: 25                                                     # Minimum Ta4 on User-Plane in microseconds.
  is_prach_cp_enabled: true                                      # Configures if Control-Plane messages should be used to receive PRACH messages.
  ignore_ecpri_payload_size: true
  compr_method_ul: bfp                                            # Uplink compression method.
  compr_bitwidth_ul: 8                                            # Uplink IQ samples bitwidth after compression.
  compr_method_dl: bfp                                            # Downlink compression method.
  compr_bitwidth_dl: 8                                            # Downlink IQ samples bitwidth after compression.
  compr_method_prach: bfp                                         # PRACH compression method.
  compr_bitwidth_prach: 8                                         # PRACH IQ samples bitwidth after compression.
  enable_ul_static_compr_hdr: true                                # Configures if the compression header is present for uplink User-Plane messages (false) or not present (true).
  enable_dl_static_compr_hdr: true                                # Configures if the compression header is present for downlink User-Plane messages (false) or not present (true).
  iq_scaling: 5.5                                                # IQ samples scaling factor applied before compression, should be a positive value smaller than 10.
  cells:
    - network_interface: enp1s0f1                                 # Ethernet interface name used to communicate with the RU.
      ru_mac_addr: 0C:83:CC:0F:E9:04                              # RU MAC address.
      du_mac_addr: 98:b7:85:1e:e7:f0                              # DU MAC address.
      vlan_tag: 564                                                 # VLAN tag value.
#      prach_port_id: [4]                                          # PRACH eAxC port value.
#      dl_port_id: [0, 1]                                          # Downlink eAxC port values.
#      ul_port_id: [0]                                          # Uplink eAxC port values.
      prach_port_id: [4, 5, 6, 7]                                 # PRACH eAxC port values.
      dl_port_id: [0]                                             # Downlink eAxC port values.
      ul_port_id: [0, 1, 2, 3]                                    # Uplink eAxC port values.

cell_cfg:
  dl_arfcn: 641676                                                # ARFCN of the downlink carrier (center frequency).
  band: 78                                                        # The NR band.
  channel_bandwidth_MHz: 100                                      # Bandwith in MHz. Number of PRBs will be automatically derived.
  common_scs: 30                                                  # Subcarrier spacing in kHz used for data.
  plmn: "31501"                                                   # PLMN broadcasted by the gNB.
  tac: 1                                                          # Tracking area code (needs to match the core configuration).
  pci: 1                                                          # Physical cell ID.
#  nof_antennas_dl: 2                                              # Downlink number of antennas.
#  nof_antennas_ul: 1                                              # Uplink number of anntennas.
  nof_antennas_dl: 1                                              # Number of transmission antennas.
  nof_antennas_ul: 1                                              # Number of reception antennas.
  prach:
#    prach_config_index: 7                                          # PRACH configuration index.
#    prach_root_sequence_index: 1                                  # PRACH root sequence index.
#    zero_correlation_zone: 0                                      # Zero correlation zone.
#    prach_frequency_start: 0                                      # Offset in PRBs of lowest PRACH transmission occasion in frequency domain respective to PRB 0.
    prach_config_index: 159                                       # PRACH configuration index.
    prach_root_sequence_index: 1                                  # PRACH root sequence index.
    zero_correlation_zone: 0                                      # Zero correlation zone.
    prach_frequency_start: 12                                     # Offset in PRBs of lowest PRACH transmission occasion in frequency domain respective to PRB 0.

log:
  filename: /var/log/srsran/gnb.log
#/home/malin/gnb.log                                          # Path of the log file.
  all_level: info                                              # Logging level applied to all layers.

pcap:
  ngap_filename: /tmp/gnb_ngap.pcap     # Optional TEXT (/tmp/gnb_ngap.pcap). Path for NGAP PCAPs. 
  ngap_enable: false                    # Optional BOOL (false). Enable/disable NGAP packet capture.
  e1ap_filename: /tmp/gnb_e1ap.pcap     # Optional TEXT (/tmp/gnb_e1ap.pcap). Path for E1AP PCAPs. 
  e1ap_enable: false                    # Optional BOOL (false). Enable/disable E1AP packet capture.
  f1ap_filename: /tmp/gnb_f1ap.pcap     # Optional TEXT (/tmp/gnb_f1ap.pcap). Path for F1AP PCAPs. 
  f1ap_enable: false                    # Optional BOOL (false). Enable/disable F1AP packet capture.
  rlc_filename: /tmp/gnb_rlc.pcap       # Optional TEXT (tmp/gnb_rlc.pcap). Path for RLC PCAPs. 
  rlc_rb_type: all                      # Optional TEXT. Sets the RLC PCAP RB type. Supported: [all, srb, drb].
  rlc_enable: false                     # Optional BOOL (false). Enable/disable RLC packet capture.  
  mac_filename: /tmp/gnb_mac.pcap       # Optional TEXT (/tmp/gnb_mac.pcap). Path for MAC PCAPs. 
  mac_type: udp                         # Optional TEXT (udp). Sets the MAC PCAP pcap type. Supported: [DLT or UDP].
  mac_enable: false                     # Optional BOOL (false). Enable/disable MAC packet capture.
  e2ap_filename: /tmp/gnb_e2ap.pcap     # Optional TEXT (/tmp/gnb_e2ap.pcap). Path for E2AP PCAPs. 
  e2ap_enable: false                    # Optional BOOL (false). Enable/disable E2AP packet capture.
  gtpu_filename: /tmp/gnb_gtpu.pcap     # Optional TEXT (/tmp/gnb_gtpu.pcap). Path for GTPU PCAPs. 
  gtpu_enable: false                    # Optional BOOL (false). Enable/disable GTPU packet capture.

```

## Expected Behavior ##
I don't exactly know what is the expected behavior but  we have tested it with zmq base architecture and this is suppose to be expected behavior:
```
2024-03-06T07:15:11.053244 [DU-F1   ] [I] F1 Setup: Sending F1 Setup Request to CU-CP...
2024-03-06T07:15:11.053330 [DU-F1   ] [I] F1 Setup: Procedure completed successfully.
2024-03-06T07:15:11.053399 [SCHED   ] [I] [     0.0] cell=0: Cell configured successfully.
2024-03-06T07:15:11.067794 [DU-MNG  ] [I] DU manager started successfully.
2024-03-06T07:15:11.067797 [DU      ] [I] DU-High started successfully
2024-03-06T07:15:11.067800 [DU      ] [I] DU started successfully
2024-03-06T07:15:11.067800 [GNB     ] [I] Starting Radio Unit...
2024-03-06T07:15:11.067820 [GNB     ] [I] Radio Unit started successfully
2024-03-06T07:15:12.067812 [zmq:tx:0:0] [I] Waiting for request.
2024-03-06T07:15:12.067849 [zmq:rx:0:0] [I] Waiting for data.
2024-03-06T07:15:12.067897 [zmq:rx:0:0] [I] Waiting for reading samples. Completed 0 of 11520 samples.
2024-03-06T07:15:13.067814 [zmq:tx:0:0] [I] Waiting for request.
2024-03-06T07:15:13.067850 [zmq:rx:0:0] [I] Waiting for data.
2024-03-06T07:15:13.067903 [zmq:rx:0:0] [I] Waiting for reading samples. Completed 0 of 11520 samples.
2024-03-06T07:15:13.769620 [GNB     ] [I] Stopping Radio Unit...

```

## Actual Behaviour ##
Instead what I am getting is this:
```
2024-03-12T05:36:04.706010 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '477.17' and sector#0
2024-03-12T05:36:04.707012 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '477.19' and sector#0
2024-03-12T05:36:04.711014 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '478.7' and sector#0
2024-03-12T05:36:04.712014 [OFH     ] [W] Missed incoming User-Plane PRACH messages for slot '478.9' and sector#0
```

## Additional Information ##
Basically I tested with a 5g device with a 5g sim. But my device couldn't find the network and this is appearing. What is the meaning of this message?? And what is the usual solution of this?

Thank you.

**Comment by aibtw** [comment]
I am no expert but I have an addition that might help someone else to understand the problem better.
I noticed that even when I was able to get time sync and **connected UE to RU successfully**, I still got `Missed incoming User-Plane PRACH` very frequently. 

So this makes me think it might be something else other than timing?

**Comment by ehitimum** [comment]
> I am no expert but I have an addition that might help someone else to understand the problem better. I noticed that even when I was able to get time sync and **connected UE to RU successfully**, I still got `Missed incoming User-Plane PRACH` very frequently.
> 
> So this makes me think it might be something else other than timing?

Is it possible for you to provide your .log result and also srsran config file so that I can compare it with mine. Also I myself wanted to connect **UE** with **RU** so it will be helpful.

Thank You for your feedback.

**Comment by andrepuschmann** [comment]
What RU are you using?
